core/dbt/node_runners.py

```diff
@@ -551,6 +551,11 @@ def execute_test(self, test):
             raise RuntimeError(
                 "Bad test {name}: Returned {rows} rows and {cols} cols"
                 .format(name=test.name, rows=num_rows, cols=num_cols))
+        elif num_rows == 0:
```

Comment:
What do you think about changing the `if` statement above this line to `if num_rows != 1`?

Reply:
I initially changed the `if` statement to use that exact logic, but ended up with the `elif` so that we could communicate a different `RuntimeError` message back to the user. 

If you think the first error message is informative enough I'm happy to make that change quickly!

Reply:
Yeah - I think that calling out the "shape" of the query results here makes sense. I suppose we can update this message to says "Bad test {name}: returned {rows} rows and {cols} cols but expected 1 row and 1 column". You buy that?

We use pep-8 as a style guide for dbt -- just make sure the log line isn't longer than 79 characters and you should be good!

Reply:
Sounds good to me, I've updated the logic, rebased, and force pushed over the branch for this PR.

Reply:
looks great!

You did a really amazing job with this PR - it was a pleasure to review :D 


Going to merge this when the tests are passing - it will ship in 0.15.0 in a couple of weeks